packages/react/src/components/header/components/headerActionBar/HeaderActionBarNavigationMenu.tsx

```diff
@@ -1,47 +1,348 @@
+  const previousLabel = link ? link.props.label : frontPageLabel;
+  return (
+    <li className={styles.previousListItem}>
+      <button type="button" className={styles.previousLinkWrapper} onClick={() => onClick(link || titleHref)}>
+        <span className={styles.previousButton}>
+          <IconAngleLeft className={styles.backIcon} />
+        </span>
+        <span className={styles.previousMobileLink}>{previousLabel}</span>
+      </button>
+    </li>
+  );
+};
+
+const ActiveDropdownLink = ({ id, link, frontPageLabel, titleHref, onLinkClick }: DropdownLinkProps) => {
+  const className = styles.activeMobileLink;
+  const activeLink = link ? (
+    cloneElement(link, {
+      id,
+      className,
+      dropdownButtonClassName: styles.hideDropdownButton,
+      wrapperClassName: styles.mobileLinkWrapper,
+      onClick: onLinkClick,
+    })
+  ) : (
+    <HeaderLink id={id} label={frontPageLabel} href={titleHref} className={className} onClick={onLinkClick} />
+  );
+  return (
+    <li className={styles.activeListItem}>
+      <span className={styles.activeLinkWrapper}>{activeLink}</span>
+    </li>
+  );
+};
+
+type MenuLinksProps = {
+  links: React.ReactNode[];
+  onDropdownButtonClick: (link) => void;
+  onLinkClick: (link) => void;
+};
+const MenuLinks = ({ links, onDropdownButtonClick, onLinkClick }: MenuLinksProps) => {
+  return (
+    <>
+      {links.map((child, index) => {
+        if (!isValidElement(child)) return null;
+        return (
+          // eslint-disable-next-line react/no-array-index-key
+          <li key={index}>
+            <span className={styles.mobileNavigationLink}>
+              {cloneElement(child as React.ReactElement, {
+                dropdownLinkClassName: styles.mobileDropdownLink,
+                dropdownClassName: styles.mobileLinkDropdown,
+                wrapperClassName: styles.mobileLinkWrapper,
+                dropdownButtonClassName: styles.mobileLinkDropdownButton,
+                className: classNames(child.props.className, styles.mobileLink),
+                index,
+                onDropdownButtonClick: () => onDropdownButtonClick(child),
+                onClick: () => onLinkClick(child),
+              })}
+            </span>
+          </li>
+        );
+      })}
+    </>
+  );
+};
+
+/**
+ * Find only the active links in the given objects model.
+ * @param obj
+ * @returns
+ */
+function findActiveLinks(obj) {
+  const activeLinks = [];
+
+  if (obj.props.active) {
+    activeLinks.push(obj);
+  }
+
+  if (obj.props.dropdownLinks && obj.props.dropdownLinks.length > 0) {
+    obj.props.dropdownLinks.forEach((link) => {
+      activeLinks.push(...findActiveLinks(link));
+    });
+  }
+
+  return activeLinks;
+}
+
+function getActiveMainLevelLink(links) {
+  return getChildrenAsArray(links).find((link) => {
+    if (isValidElement(link) && link.props.active) return link;
+    return null;
+  });
```

Comment:
I think this would do the same thing 🤔 
```
 return getChildrenAsArray(links).find((link) => {
    return isValidElement(link) && link.props.active;
});
```